ensenso_camera/src/virtual_object_handler.cpp

```diff
@@ -41,7 +41,7 @@ struct VirtualObjectMarkerPublisher
   VirtualObjectMarkerPublisher(ensenso::ros::NodeHandle _nh, const std::string& topic, double publishRate,
                                const NxLibItem& objects, const std::string& frame, std::atomic_bool& stop_)
```

Comment:
Getting the following error when using the suggested `updateObjectLinks()` function:
`src/ros_driver/ensenso_camera2/src/virtual_object_handler.cpp:205:27: error: expected primary-expression before ‘[’ token
  205 |   auto objects = NxLibItem[itmObjects];`

Reply:
Sorry, that must be `NxLibItem()[itmObjects];`

Reply:
Got the following error now after using `NxLibItem()[itmObjects];`:
`NxLibException 13 (ItemTypeNotCompatible) for item /Objects/\0/Link/Target`

Reply:
The code looks good. The error you're getting (see [here](https://www.ensenso.com/manual/3.4/api/return-codes.html?highlight=itemtypenotcompatible)) says that you are trying to get a value as - I guess - a string, but the value in the NxLib tree is not a string. Could it be that the target frames in the objects file your are testing with are `null`? At least they are when exporting an objects file from NxView.

Reply:
Maybe we should still pass the `objectsFrame` value to the `VirtualObjectHandler` and let it be used as a default value for objects that do not have a specified `target` frame?

Reply:
Yes we could do that. However I do not see any `null` values in my objects file, could it be that NxTree does not set the `Target` to any value?

Reply:
Strange. I tested it on my system and I had no problems. Since I couldn't push to your remote branch I created a new one called "VirtualObjectFrames" in our repo. It contains your two commits and another one from me with debug output and my objects file.

The error you are getting is referring to which line of code?

Reply:
Line 213

Reply:
Which version of Ensenso SDK do you use?
I am using 3.3.1417

Reply:
I just tried it with both 3.3.1417 and 3.4.743, both with the same results.

In line 168 you write the objects to the NxLib tree. From there on they are in the tree. Getting the target name in line 213 should work, unless you are modifying the tree from somewhere else? Are you perhaps adding or removing objects in the meantime? The error could e.g. occur if you added another object meanwhile, which has no target node. 

In case you want to be able to add and remove objects during operation, storing the object frames as you initially did would be required.

Reply:
Yes I do add other objects in the meantime. I have two `ensenso_camera_node` nodes each with their own virtual objects file. They are both started from the same launch file.

Reply:
Okay I see. Then there is the problem.

Since you guys are currently the only one developing (and maybe even using) this feature, it is quite hard for me - or us in general - to review the code. Do you think you could maybe write a test case that also acts as kind of a documentation? Then I can get an idea of how your code is supposed to work.